includes/abstracts/abstract-wc-product.php

```diff
@@ -1145,13 +1145,15 @@ public function set_downloads( $downloads_array ) {
 				$download_object = $download;
 			} else {
 				$download_object           = new WC_Product_Download();
-				$download['previous_hash'] = isset( $download['previous_hash'] ) ? $download['previous_hash'] : '';
-				$file_hash                 = apply_filters( 'woocommerce_downloadable_file_hash', md5( $download['file'] ), $this->get_id(), $download['name'], $download['file'], $download['previous_hash'] );
 
-				$download_object->set_id( $file_hash );
+				// If we don't have a previous hash, generate UUID for download
```

Comment:
We don't have phpcs enabled on here yet (holding off until 3.3) but just wanted to point out these types of comments should end with a full stop. &lt;/nitpick&gt; :)